### DISCUSSION TITLE
Copying Headers/Baggage to Span Attributes for Collection

### DISCUSSION THREAD

**Discussion by paychex-ssmithrand** [open-telemetry/opentelemetry-ruby#1360] [Q&A] [answered]
Your favorite Otel rookie here again! 

There are some custom headers that I'm adding to baggage that I'd like to also set as span attributes, which will make their way to the collector. Should I be doing this within a propagator or in a span processor? If a span processor, is there a way to chain them similarly to how the composite propagator works? I'd like to keep using the BatchSpanProcessor and would hate to have to re-create it from scratch with some minute logic to add baggage as attributes.

**Comment by arielvalentin** [comment, marked as answer]
See this related discussion https://github.com/open-telemetry/opentelemetry-ruby/discussions/1272
